File: app/src/main/java/eu/napcode/gonoteit/ui/main/MainViewModel.java
package eu.napcode.gonoteit.ui.main;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import javax.inject.Inject;

import eu.napcode.gonoteit.model.UserModel;
import eu.napcode.gonoteit.repository.user.UserRepository;

public class MainViewModel extends ViewModel {

    private UserRepository userRepository;
    private MutableLiveData<UserModel> loggedInUser = new MutableLiveData<>();

    @Inject
    public MainViewModel(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public MutableLiveData<UserModel> getLoggedInUser() {
        loggedInUser.postValue(userRepository.getLoggedInUser());

        return loggedInUser;
    }

    public void logoutUser() {
        userRepository.logoutUser();

        getLoggedInUser();
    }
}

File: app/src/main/java/eu/napcode/gonoteit/ui/main/MainViewModel.kt
package eu.napcode.gonoteit.ui.main

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel

import javax.inject.Inject

import eu.napcode.gonoteit.model.UserModel
import eu.napcode.gonoteit.repository.user.UserRepository
import eu.napcode.gonoteit.rx.RxSchedulers

class MainViewModel @Inject
constructor(private val userRepository: UserRepository, private val rxSchedulers: RxSchedulers) : ViewModel() {

    fun getLoggedInUser(): LiveData<UserModel?> {
        return userRepository.getLoggedInUser()
    }

    fun logoutUser() {
        userRepository
                .logoutUser()
                .subscribeOn(rxSchedulers.io())
                .observeOn(rxSchedulers.androidMainThread())
                .doFinally { getLoggedInUser() }
                .subscribe()
    }

    fun isUserLoggedIn() : Boolean {
        return userRepository.isUserLoggedIn()
    }
}
